src/query-compiler/default-query-compiler.ts

```diff
@@ -798,6 +798,10 @@ export class DefaultQueryCompiler
     }
 
     if (node.joins) {
+      if (!node.from) {
+        throw new Error("Joins in an update query are only supported as a part of a PostgreSQL 'update set from join' query. If you want to create a MySQL 'update join set' query, see https://kysely.dev/docs/examples/update/my-sql-joins")
```

Comment:
nitpicky

maybe we should expose a `KyselyError`?

Reply:
Maybe, but I'd add that as a separate PR since we should use that everywhere if we so decide.